db/import_mb_data.py

```diff
@@ -0,0 +1,1512 @@
+            VALUES (:id, :gid, :name, :artist_credit, :release_group, :status, :packaging, :language,
+                    :script, :barcode, :comment, :edits_pending, :quality, :last_updated)
+    """)
+    values = [{
+        "id": value[0],
+        "gid": value[1],
+        "name": value[2],
+        "artist_credit": value[3],
+        "release_group": value[4],
+        "status": value[5],
+        "packaging": value[6],
+        "language": value[7],
+        "script": value[8],
+        "barcode": value[9],
+        "comment": value[10],
+        "edits_pending": value[11],
+        "quality": value[12],
+        "last_updated": value[13]} for value in MB_release_data
+    ]
+    connection.execute(release_query, values)
+    transaction.commit()
+    print("INSERTED release data\n")
+
+
+def write_release_gid_redirect(transaction, connection):
+    release_gid_redirect_query = text("""
+        INSERT INTO musicbrainz.release_gid_redirect
+            VALUES (:gid, :new_id, :created)
+    """)
+    values = [{
+        "gid": value[0],
+        "new_id": value[1],
+        "created": value[2]} for value in MB_release_gid_redirect_data
+    ]
+    connection.execute(release_gid_redirect_query, values)
+    transaction.commit()
+    print("INSERTED release_gid_redirect data\n")
+
+
+def write_medium(transaction, connection):
+    medium_query = text("""
+        INSERT INTO musicbrainz.medium
+            VALUES (:id, :release, :position, :format, :name, :edits_pending, :last_updated, :track_count)
+    """)
+    values = [{
+        "id": value[0],
+        "release": value[1],
+        "position": value[2],
+        "format": value[3],
+        "name": value[4],
+        "edits_pending": value[5],
+        "last_updated": value[6],
+        "track_count": value[7]} for value in MB_medium_data
+    ]
+    connection.execute(medium_query, values)
+    transaction.commit()
+    print("INSERTED medium data\n")
+
+
+def write_track(transaction, connection):
+    track_query = text("""
+        INSERT INTO musicbrainz.track
+            VALUES (:id, :gid, :recording, :medium, :position, :number, :name, :artist_credit, :length,
+                    :edits_pending, :last_updated, :is_data_track)
+    """)
+    values = [{
+        "id": value[0],
+        "gid": value[1],
+        "recording": value[2],
+        "medium": value[3],
+        "position": value[4],
+        "number": value[5],
+        "name": value[6],
+        "artist_credit": value[7],
+        "length": value[8],
+        "edits_pending": value[9],
+        "last_updated": value[10],
+        "is_data_track": value[11]} for value in MB_track_data
+    ]
+    connection.execute(track_query, values)
+    transaction.commit()
+    print("INSERTED track data\n")
+
+
+
+
+# FUNCTION TO CALL ALL INSERTS
+
+def insert_MB_data_AB():
+    with db.engine.connect() as connection:
+        if MB_artist_credit_data:
+            transaction = connection.begin()
```

Comment:
Do we need a transaction for each insert?

Reply:
Removed the integrity check part as we are using 'on conflict action'.